org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/StandardProjectsManager.java

```diff
@@ -668,6 +668,25 @@ public void projectsImported(IProgressMonitor monitor) {
 		this.preferenceManager.getPreferences().updateAnnotationNullAnalysisOptions();
 	}
 
+	@Override
+	public void cleanInvalidJavaProjects(IProgressMonitor monitor) {
```

Comment:
Could this logic be placed into the `IProjectManager.cleanInvalidProjects(..)` (for `StandardProjectsManager`) instead creating a separate method interface ? I noticed that whole logic seems to be mainly concerned with gradle projects currently. Or is it not possible because that logic is called on `initialize` and you need your stuff to run on `initialized` ?

Reply:
I need to run it on initialized.

Reply:
Would it make sense to have it just be a separate static method since we're only calling it once in one specific area ?

Reply:
Fixed.